src/Campaigns/Blocks/shared/components/ModalForm/index.tsx

```diff
@@ -27,6 +28,19 @@ export default function ModalForm({dataSrc, embedId, buttonText, isFormRedirect,
     const [isLoading, setLoading] = useState<boolean>(false);
     const [isEntering, setEntering] = useState<boolean>(false);
 
+    useEffect(() => {
+        const handleEscape = (event: KeyboardEvent) => {
+            if (event.key === 'Escape' && isOpen) {
+                closeModal();
+            }
+        };
+
+        document.addEventListener('keydown', handleEscape);
```

Comment:
Technically the modal component is supposed to[ suppose the esc button](https://react-spectrum.adobe.com/react-aria/Modal.html#interactions) but I think it stopped working reliably once I overloaded the entering prop.

I think this is simple enough to keep to compensate